src/clarity-angular/input-trap/input-trap.directive.ts

```diff
@@ -0,0 +1,47 @@
+import { Directive, HostListener, ElementRef, OnInit } from "@angular/core";
+
+@Directive({
+    selector: "[clrInputTrap]"
+})
+export class InputTrapDirective implements OnInit {
```

Comment:
This should be named "Focus trap". 😉 
You could very well have a modal with no inputs, just buttons, that wants to prevent focus outside.